src/jsi/jsi.cpp

```diff
@@ -360,6 +385,8 @@ JSError::JSError(Runtime& rt, std::string msg, std::string stack)
     e.setProperty(rt, "message", String::createFromUtf8(rt, message_));
     e.setProperty(rt, "stack", String::createFromUtf8(rt, stack_));
     setValue(rt, std::move(e));
+  } catch (const std::exception& ex) {
```

Comment:
>catch [](start = 4, length = 5)

on the overall design. Why are we moving toward exception based model? This pattern feels very tricky thus could be hard to get right. Would it make sense to go with model of something like:

bool Foo(args, Error& errorMessage) noexcept {...}